### PR TITLE
test_models: panda safety tx test

### PR DESCRIPTION
**Pull request by sshane**
sanity checks on the panda safety for sending standard openpilot messages.

asserts:
- [x] We send all messages with empty CC message (nothing should be active)
- [x] Cancel message works (in correct state)
- [x] Resume message works

### CODE REVIEW

File: selfdrive/car/tests/test_models.py [hunk 1]
```diff
@@ -201,6 +201,42 @@ def test_panda_safety_rx_valid(self):
           self.assertTrue(self.safety.addr_checks_valid())
     self.assertFalse(len(failed_addrs), f"panda safety RX check failed: {failed_addrs}")
 
+  def test_panda_safety_tx_cases(self, data=None):
+    """Asserts we can tx common messages"""
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:205]: why is notCar skipped?

File: selfdrive/car/tests/test_models.py [hunk 2]
```diff
@@ -201,6 +201,42 @@ def test_panda_safety_rx_valid(self):
...
+    if self.CP.notCar:
+      self.skipTest("Skipping test for notCar")
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:207]: why are not cars skipped?

File: selfdrive/car/tests/test_models.py [hunk 3]
```diff
@@ -201,6 +201,42 @@ def test_panda_safety_rx_valid(self):
...
+        _, sendcan = self.CI.apply(car_control, _now_nanos)
+        _now_nanos += 10e6
+        msgs_sent += len(sendcan)
+        for addr, _, dat, bus in sendcan:
+          to_send = libpanda_py.make_CANPacket(addr, bus % 4, dat)
+          self.assertTrue(self.safety.safety_tx_hook(to_send), (addr, dat, bus))
+
+      # Make sure we attempted to send messages
+      self.assertGreater(msgs_sent, 50)
+      return _now_nanos
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:222]: no need to return or pass this in, just use a nonlocal

File: selfdrive/car/tests/test_models.py [hunk 4]
```diff
@@ -201,6 +201,42 @@ def test_panda_safety_rx_valid(self):
...
     self.assertFalse(len(failed_addrs), f"panda safety RX check failed: {failed_addrs}")
 
+  def test_panda_safety_tx_cases(self, data=None):
+    """Asserts we can tx common messages"""
+    if self.CP.notCar:
+      self.skipTest("Skipping test for notCar")
+
+    def test_car_controller(car_control, _now_nanos):
+      msgs_sent = 0
+      for _ in range(300):  # make sure we hit the slowest messages
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:211]: use DT_CTRL, would also probably make this run longer. maybe 10s?

File: selfdrive/car/tests/test_models.py (hunk 3 above)
Comment (adeebshihadeh) [reply selfdrive/car/tests/test_models.py:222]: or even better, make a new self.CI so that it's scope is just within the function

**Review by adeebshihadeh — APPROVED** [review]

**Comment by adeebshihadeh** [comment]
Looks good, just needs some cleanup

File: selfdrive/car/tests/test_models.py (hunk 2 above)
Comment (sshane) [reply selfdrive/car/tests/test_models.py:207]: openpilot and safety both assume that `controls_allowed` will always effectively be true. if it's false, it blocks all messages, even zero torque ones, so we need to fix that first before enabling

**Pull request merged**